### ISSUE TITLE
Separate DeploymentPlan creation from its execution

### ISSUE THREAD

**Issue by PaulFarault** [TOSIT-IO/tdp-lib#318]
### Issue

The `deploy` command automatically execute a deployment plan. Each operation result is inserted in the database sequentially. When an operation fails, the following operations are skipped and are not persisted in the database. So, we can't be sure that resuming a failed deployment effectively execute the initially planed operations.

### Proposition

We should be able to separate the planification from the execution. Hence, to persist the whole deployment plan in the database before launching it.

- The `plan` command must save a deployment plan with all of its operations in the database.
- The `deploy` command should only launch the planed deployment.
- If no deployment is planed, the command raises an error.
- We can't plan 2 deployments in parallel. If more than 1 planed deployment exists in the database, an error is raised when doing the deploy.
- Creating a new plan or modifying one will override the initial one.
- The `resume` deployment mode starts from the failed operation if the last deployment is a failure.
- The `reconfigure` deployment mode can be treated later (see #308)
- Planed deployment should be edited (to do later)

### Implementation

We will need to:

- [x] Add new value to the status enum
- [x] Update the database schema
- [x] Write the whole deployment plan in the database before execution
- [x] Rewrite the execution Iterator
- [x] Rewrite the logic of the plan, resume and deploy commands

#### Database schema

Add an `order` column to the `operation_logs`. Maybe rename the `operation_log` to `operation` table.

#### Status enum

The following Operation status can be considered:

- **PLANNED**, deployment is not launched, operation can be modified.
- **PENDING**, deployment is launched, operation is waiting to be executed.
- **RUNNING**, operation is being executed.
- **SUCCESS**/**FAILURE**, operation succeeded/failed.
- **HELD**, the operation hasn't been executed (a previous one has been paused or failed).

Lifecycle example:

- Planification

  ```
  PLANNED
  PLANNED
  PLANNED
  PLANNED
  ```

- Execution

  ```
  SUCCESS
  RUNNING
  PENDING
  PENDING
  ```

- In case of a failure

  ```
  SUCCESS
  FAILURE
  HELD
  HELD
  ```

**Comment by sergkudinov** [comment]
in my opinion, a single `plan` command with multiple options

**Comment by rpignolet** [comment]
We can use subcommand too. IMO, it depends on arguments. If they have different arguments we should use subcommand for example `tdp plan dag`, `tdp plan run`, `tdp plan reconfigure`, `tdp plan resume`.
If they all have the same arguments just one `tdp plan`.

**Comment by PaulFarault** [comment]
This will be subcommands then. Thanks!
